### PR TITLE
Allow importing when there is a dot in a folder name

### PR DESCRIPTION
**Pull request by alanhogan**
This change may yield some number of somewhat nonsensical searched filenames in some cases (like `.scss.scss`) but it also allows imports like `my.folder/my.styles` I believe it may also help with Asset Pipeline-influenced names like `.css.scss`.

### CODE REVIEW

**Comment by alanhogan** [comment]
Note this PR is a strict superset of #56

**Comment by alanhogan** [comment]
@chriseppstein I ran into a lot of file-not-found sort of errors when moving a big project from ruby sass / compass. All these changes helped!

Nasty patterns that were essentially out of my control, Sass-wise:

- Styles shared via `bower_components` with dots in the bower module name (`shared.styles/font.weights`)
- Sass files named in an Asset Pipeline-influenced way: `shared-styles.css.scss`

In short, this PR helps find more of the files that might otherwise go unfound. And it doesn’t break any existing tests :)
